xla/stream_executor/stream_executor.h

```diff
@@ -177,7 +177,10 @@ class StreamExecutor {
       uint64_t size) = 0;
 
   // Deallocates a region of host memory allocated by HostMemoryAllocate().
-  virtual void HostMemoryDeallocate(void* mem) = 0;
+  [[deprecated(
+      "Use HostMemoryDeallocate(void*, uint64_t) instead")]] virtual void
+  HostMemoryDeallocate(void* mem) = 0;
```

Comment:
Couldn't we just remove this deprecated overload?

Reply:
This dates back to a comment on the previous PR (https://github.com/openxla/xla/pull/15216#discussion_r1694888936) that the deprecated version was still used in other dependent projects, such as tensorflow.

However, since you reviewed last week @klucke landed some other commits (17a05c7a8628e8ba4e97fe31c1d70bc8b65f8439, ef40c8bd7d0701e06bea299c87a2ede4e09e42d2, 9a03576e2811a7abad6f1ce98fad146e84700e74) that changed how deallocation has handled, so this part of the PR now looks a bit different and less invasive.